src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -145,8 +149,10 @@ def get_agent_version(self):
                 WALinuxAgent-2.2.49.2 running on sles 15.3
                 Python: 3.6.13
                 Goal state agent: 2.6.0.2 '''
-            return self.__agent_version_string_search(r'WALinuxAgent-\S+ running', out)
+            return self.__extract_agent_version_from_string(r'WALinuxAgent-\S+ running', out)
 
+        # Command failed, so log error and debugging information
+        self.composite_logger.log_error('Failed to execute command to get guest agent version. [Code={}] [Out=\'{}\']'.format(str(code), str(out)))
```

Comment:
While non-indexed placeholders work, the pattern we follow in the code base is to index them for readability. Please adjust all the pre-format strings to align.

Reply:
Thanks, fixed!